src/engine/polyline_compressor.cpp

```diff
@@ -61,24 +61,25 @@ std::vector<util::Coordinate> decodePolyline(const std::string &geometry_string)
 
     while (index < len)
     {
-        int b, shift = 0, result = 0;
-        do
+        int b = 0x20, shift = 0, result = 0;
+        while (b >= 0x20 && index < len)
```

Comment:
can you explain if the switch here does change anything in the order/count of execution? The first loop looks like it should be executed exactly as often as before. I am totally fine with the change, I just wonder if there is a reason other than style?